### ISSUE TITLE
"Undo" and "Redo" action should apply to a whole multicommand instead of each child action in it

### ISSUE THREAD

**Issue by TMPxyz** [ryuta46/vscode-multi-command#49]
If we execute a multicmd  consisting of five "cursorDown" sub-actions, then press "ctrl-z" should undo the whole multicmd instead of a single "cursorDown".

**Comment by Bobronium** [comment]
I'm new to VSCode, but looking at what extension does, It'll probably require some trickery like remembering the number of executed actions, monitoring current state in history, hijacking Undo shortcut and upon its activation execute some logic depending on current context.

Sounds like a hassle, but I really hope I'm just missing something. @ryuta46, do you have any better insights or hints on this?

**Comment by ryuta46** [comment]
Indeed, being able to undo the entire sequence of a multi-command with a single Undo does sound intuitive and appealing.
However, at the moment it is difficult to achieve this for the following reasons:
* Cursor movements such as `cursorDown` are not added to VS Code’s undo stack, so they cannot be undone in the usual way. Only edits to the editor (like text input) are pushed onto the undo stack; navigation operations such as switching tabs are not. The extension cannot distinguish between these kinds of commands.
* The undo stack is managed per editor. Therefore, unless undo operations are performed while navigating appropriately, it is not possible to restore the original state. It is very difficult for an extension to take the full content of the commands into account and perform such an undo.

Even if implemented, the functionality would inevitably be limited to undoing consecutive edit commands within a single editor, which doesn’t seem particularly useful.
